src/TraceEvent/Symbols/R2RPerfMapSymbolModule.cs

```diff
@@ -0,0 +1,246 @@
+            {
+                throw new FormatException($"Could not parse size from '{sizeStr}'");
+            }
+
+            // Get the name (everything after the second space)
+            string name = remainder.Substring(secondSpaceIndex + 1);
+
+            return new R2RPerfMapSymbol(address, size, name);
+        }
+
+        private void OnParsedSymbol(R2RPerfMapSymbol symbol)
+        {
+            switch (symbol.StartAddress)
+            {
+                case 0xFFFFFFFF:
+                    if (Guid.TryParse(symbol.Name, out Guid signature))
+                    {
+                        Signature = signature;
+                    }
+                    else
+                    {
+                        throw new FormatException($"Could not parse signature from '{symbol.Name}'");
+                    }
+                    break;
+                case 0xFFFFFFFE:
+                    uint parsedVersion;
+                    if (uint.TryParse(symbol.Name, out parsedVersion))
+                    {
+                        Version = parsedVersion;
+                    }
+                    else
+                    {
+                        throw new FormatException($"Could not parse version from '{symbol.Name}'");
+                    }
+                    break;
+                case 0xFFFFFFFD:
+                    if (Enum.TryParse(symbol.Name, out R2RPerfMapOS os) && os < R2RPerfMapOS.Invalid)
+                    {
+                        TargetOS = os;
+                    }
+                    else
+                    {
+                        throw new FormatException($"Could not parse OS from '{symbol.Name}'");
+                    }
+                    break;
+                case 0xFFFFFFFC:
+                    if (Enum.TryParse(symbol.Name, out R2RPerfMapArchitecture arch) && arch < R2RPerfMapArchitecture.Invalid)
+                    {
+                        TargetArchitecture = arch;
+                    }
+                    else
+                    {
+                        throw new FormatException($"Could not parse architecture from '{symbol.Name}'");
+                    }
+                    break;
+                case 0xFFFFFFFB:
+                    if (Enum.TryParse(symbol.Name, out R2RPerfMapABI abi) && abi < R2RPerfMapABI.Invalid)
+                    {
+                        TargetABI = abi;
+                    }
+                    else
+                    {
+                        throw new FormatException($"Could not parse ABI from '{symbol.Name}'");
+                    }
+                    break;
+                default:
+                    _symbols.Add(symbol);
+                    break;
+            }
+        }
+
+        private void FinalizeSymbols()
+        {
+            _symbols.Sort((a, b) => a.StartAddress.CompareTo(b.StartAddress));
+        }
+
+        private int BinarySearch(uint rva)
```

Comment:
Array.BinarySearch() instead?

Reply:
Thanks @noahfalk!  I looked into this, but it looks like you have to give it an element that matches the type of the array to compare to, so it's a bit unwieldy because we'd have to construct symbol a partial symbol object with the address of interest and then know how to compare it.  Ultimately, this seems a bit cleaner, but if I'm misunderstanding how Array.BinarySearch works, please let me know as it is exposed on List<T>, which is the type being used here.

Reply:
Yeah, I took a closer look at the interface Array.BinarySearch() provides and I agree its not an exact fit. You could make it work as you said with a partially constructed element + an IComparable implementation + some extra logic to deal with non-exact matches and range overlap checks but at that point its around the same complexity as what you already have.